0006-sync-iteration/README.md

```diff
@@ -0,0 +1,197 @@
+
+Re-synchronization can require fetching and processing a full repo CAR file, which is a relatively expensive operation. See the "Sync Boundary" section for details of this process, and "Streaming CAR Processing" for ways to make it cheaper for consumers. To prevent network abuse and avoid resource consumption, services should keep counters and rate-limit re-synchronization attempts. Accounts which are in a rate-limited state (for exceeding this or other limits) can use the new `throttled` account state.
+
+Relays do not perform the re-synchronization process. They should drop invalid commit messages (not pass-through). They should keep track of discontinuous commits, and may set the `throttled` state if limits are exceeded. Otherwise they should pass through discontinuous commits as-is (and update their "last commit" state).
+
+
+## Sync Boundary and Record Operation Stream
+
+Not all services working with atproto data need to implement "Authenticated Transfer" and verify synchronization. They might skip some aspects of verification, or use an un-authenticated event stream (like Jetstream).
+
+The intent of Authenticated Transfer is to enable interoperation (possibly adversarial interoperation) between service providers who have no relationship, all over the open internet. This authenticated synchronization often occurs indirectly, across multiple hops and providers. There is a "sphere of authenticated transfer" which starts at the account's repository host (eg, PDS), which signs commits. It terminates at the last service in the transfer chain which implements full validation. Intermediate services, like a relay, might do validation, but that is not integral or required for their role, and if they are operated by an untrusted party, they are not the "trust boundary".
+
+The "trust boundary" also corresponds with a "sync boundary": the terminating service is also responsible for re-synchronizing accounts via a `getRepo` call.
+
+This service also often unpacks commits into individual record operations. Commits can contain multiple record operations, but many applications process and index records individually. An example of this pattern is Jetstream, which terminates the sync protocol by consuming commit events, and emits record operation events. This is also the step where many applications filter out irrelevant records.
+
+Application indices (eg, AppView) often want to ensure they have processed all relevant record operations for a repository. There is a relatively efficient process to keep track, including resynchronization events:
```

Comment:
Maybe worth a link to "Streaming CAR Processing" somewhere around here?

Reply:
:heavy_check_mark: